homeassistant/components/glinet/config_flow.py

```diff
@@ -0,0 +1,186 @@
+"""Config flow for GL-inet integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from gli_py import GLinet
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components.device_tracker.const import (
+    CONF_CONSIDER_HOME,
+    DEFAULT_CONSIDER_HOME,
+)
+from homeassistant.const import CONF_API_TOKEN, CONF_HOST, CONF_PASSWORD
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.device_registry import format_mac
+
+from .const import DOMAIN
+
+# from homeassistant.helpers import config_validation as cv
+
+
+_LOGGER = logging.getLogger(__name__)
+
+# TODO adjust the data schema to the data that you need
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST, default="http://192.168.8.1"): str,
+        vol.Required(CONF_PASSWORD, default="goodlife"): str,
+    }
+)
+
+
+class TestingHub:
+    """Testing class to test connection and authentication."""
+
+    def __init__(self, host: str) -> None:
+        """Initialize."""
+        self.host: str = host
+        self.router: GLinet = GLinet(base_url=self.host + "/cgi-bin/api/", sync=False)
+        self.router_model: str = ""
+        self.router_mac: str = ""
+
+    async def connect(self) -> bool:
+        """Test if we can communicate with the host."""
+        try:
+            res = await self.router.router_model()
+            self.router_model = res["model"]
+            return True
+        except ConnectionError:
+            _LOGGER.error(
+                "Failed to connect to %s, is it really a GL-inet router?", self.host
+            )
+        except TypeError:
+            _LOGGER.error(
+                "Failed to parse router response to %s, is it the right firmware version?",
+                self.host,
+            )
+        return False
+
+    async def authenticate(self, password: str) -> bool:
+        """Test if we can authenticate with the host."""
+        try:
+            await self.router.async_login(password)
+            res = await self.router.router_mac()
+            self.router_mac = res["factorymac"]
+        except ConnectionRefusedError:
+            _LOGGER.error("Failed to authenticate with Gl-inet router during testing")
+        return self.router.logged_in
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+    # TODO validate the data can be used to set up a connection.
+
+    # If your PyPI package is not built with async, pass your methods
+    # to the executor:
+    # await hass.async_add_executor_job(
+    #     your_validate_func, data["username"], data["password"]
+    # )
```

Comment:
I think we should handle 'todos' differently, maybe just add it instead of todo-ing it?
```suggestion
```